### ISSUE TITLE
Is there a way to automate `Publish>Deploy as API executable...`

### ISSUE THREAD

**Issue by lastobelus** [danthareja/node-google-apps-script#32]
Currently when working on REST script execution targets, after running `gulp upload-latest --env dev` I have to go to the script project in a browser and run `Publish>Deploy as API executable...`, and name the version.

Is there any way to automate this as part of pushing the code up?

**Comment by oshliaer** [comment]
Is there an API for this?

**Comment by peterherrmann** [comment]
There was but it was [deprecated](https://developers.google.com/apps-script/reference/script/service#enable%28Service.Restriction%29). There is no API for this currently.

**Comment by hess-g** [comment]
Actually, that deprecated method doesn't relate to the API Executable. 

Currently, there are no APIs for any of the functions that you need to perform in the script editor, such as publishing (web apps, add-ons, API executables).

**Comment by peterherrmann** [comment]
You're correct of course. My comment referenced `Publish>Deploy as web app...` and the OP asked about `Publish>Deploy as API executable...`.

**Comment by lastobelus** [comment]
Is this still true? Constantly having to navigate the `Publish>Deploy as API executable...` interface really makes working on google app scripts a drag.

**Comment by gunar** [comment]
Related issue in Google Code: [API to deploy google apps script as WebApp](https://code.google.com/p/google-apps-script-issues/issues/detail?id=6149). But no answer from Google yet.

**Comment by shabegom** [comment]
Don't actually know if this is helpful for your use case but I have `devMode: true` set and it allows the remote execution API to run the latest version of the script without republishing. It only does this for the developer, not users so is only good for testing.

```
script.scripts.run({
        auth: auth,
        resource: {
            function: 'someFunction',
            devMode: true
        },
```

**Comment by asadmkhan** [comment]
Hi !

@lastobelus @oshliaer @peterherrmann @hess-g @gunar @sbgood 

 **I have a google sheet containing script, I want to execute this script programmatically for any user (public). Can I perform  "Publish>Deploy as API executable" programmatically?**

**Comment by oshliaer** [comment]
Hi @asadmkhan 

It's not possible.

**Comment by lricoy** [comment]
HI. Any updates on this? I've followed the links but it seems everything is still on the same page

**Comment by lricoy** [comment]
Closing as it looks idle. Please reopen if interested again.

**Comment by gunar** [comment]
@lricoy yes, we are all very much interested still!
